### PR TITLE
Fix: avoid panic on nil pointer

### PR DESCRIPTION
**Pull request by 0x656b694d**
Debugging incident-acscs-dec-18.

### CODE REVIEW

**Review by kurlov — APPROVED** [review]

File: internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go [hunk 1]
```diff
@@ -64,6 +64,21 @@ func (k *DinosaurRoutesCNAMEManager) Reconcile() []error {
...
+						glog.Infof("ChangeDinosaurCNAMErecords returned nil value")
+					case changeOutput.ChangeInfo == nil:
+						glog.Infof("ChangeDinosaurCNAMErecords returned nil ChangeInfo value")
+					case changeOutput.ChangeInfo.Id == nil:
+						glog.Infof("ChangeDinosaurCNAMErecords returned nil ChangeInfo.Id value")
+					case changeOutput.ChangeInfo.Status == nil:
+						glog.Infof("ChangeDinosaurCNAMErecords returned nil ChangeInfo.Status value")
+					}
+
+					errs = append(errs, errors.New("empty change output"))
```
Comment (kurlov) [inline internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:79]: Could you please add central ID or Name here? Or both

**Review by porridge — APPROVED** [review]

File: internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go (hunk 1 above)
Comment (porridge) [reply internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:79]: It is logged above at the top of the for loop.

File: internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go [hunk 2]
```diff
@@ -65,6 +65,11 @@ func (k *DinosaurRoutesCNAMEManager) Reconcile() []error {
 					continue
 				}
 
+				if changeOutput == nil {
```
Comment (kovayur) [inline internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:68]: nit: I'd also check `changeOutput.ChangeInfo` for `nil` to be 100% sure.

Comment (0x656b694d) [reply internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:68]: And then the Status and the Id? Well, alright, I'll bring these checks back. It's safer to be safe.

File: pkg/client/aws/client.go [hunk 3]
```diff
@@ -139,25 +138,9 @@ func (client *awsClient) ChangeResourceRecordSets(dnsName string, recordChangeBa
 
 	recordSetsOutput, err := client.route53Client.ChangeResourceRecordSets(recordChanges)
 
+	err = wrapAWSError(err, "Failed to get DNS zone.")
```
Comment (kovayur) [inline pkg/client/aws/client.go:141]: I believe we don't need to `wrapAWSError` anymore. But TBH I have a hard time understanding why this was done.

**Review by kovayur — APPROVED** [review]

File: internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go [hunk 4]
```diff
@@ -65,6 +65,15 @@ func (k *DinosaurRoutesCNAMEManager) Reconcile() []error {
 					continue
 				}
 
+				switch {
```
[the remark this answers is not in the document]
Comment (ebensh) [reply internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:68]: I understand if we want to leave it where it is to get the fix in though.

[the remark this answers is not in the document]
Comment (0x656b694d) [reply internal/dinosaur/pkg/workers/dinosaurmgrs/dinosaurs_routes_cname_mgr.go:68]: Let's leave it for now, as this should normally not be executed after the fix in the aws client.

**Pull request merged**